screens/ChooseLocationScreen.tsx

```diff
@@ -0,0 +1,274 @@
+      longitudeDelta: 0,
+    });
+  };
+
+  useEffect(() => {
+    setSelectedLocation(route.params.location);
+    if (route.params.location) {
+      animateMapToRegion(route.params.location);
+    }
+  }, [route.params.location]);
+
+  useEffect(() => {
+    const load = async () => {
+      setIsLoading(true);
+      const { success, hotspots = [] } = await loadHotspots();
+      setIsLoading(false);
+      if (!success) alert('Failed to load hotspots');
+      setHotspots(hotspots);
+    };
+    load();
+  }, []);
+
+  return (
+    <>
+      <SecondaryAppbar onBackPressed={() => navigation.goBack()} />
+      <View style={styles.mapContainer}>
+        <MapView
+          ref={mapRef}
+          // use intial region + animateToRegion instead of region as react state because
+          // gestures don't work well https://github.com/react-native-maps/react-native-maps/issues/3639
+          initialRegion={intialRegion}
+          showsUserLocation
+          style={styles.map}
+          onPress={async e => {
+            const { latitude, longitude } = e.nativeEvent.coordinate;
+            const isSameMarker =
+              selectedLocation?.longitude === longitude &&
+              selectedLocation?.latitude === latitude;
+
+            if (!isSameMarker) {
+              const location = await findPlace(latitude, longitude);
+              setSelectedLocation(location);
+            }
+          }}
+        >
+          {hotspots.map(h => (
+            <Marker
+              key={`${h.latitude} ${h.longitude}`}
+              coordinate={{
+                latitude: h.latitude,
+                longitude: h.longitude,
+              }}
+              onPress={() => navigation.navigate('Modal')}
+            >
+              <Image
+                source={getHotspotMarker(h)}
+                style={styles.marker}
+                resizeMode="contain"
+              />
+            </Marker>
+          ))}
+
+          {selectedLocation ? (
+            <Marker
+              key={`${selectedLocation.latitude} ${selectedLocation.longitude}`}
+              coordinate={{
+                latitude: selectedLocation.latitude,
+                longitude: selectedLocation.longitude,
+              }}
+              onPress={() => {
+                setSelectedLocation(defaultLocation);
+              }}
+            />
+          ) : null}
+        </MapView>
+        <TextInput
+          multiline={true}
+          outlineColor={theme.colors.disabled}
+          mode="outlined"
+          style={styles.searchInput}
+          autoCorrect={false}
+          placeholder="Caută"
+          autoComplete={false}
+          right={
+            <TextInput.Icon name="magnify" color={theme.colors.placeholder} />
+          }
+          returnKeyType="search"
+          onSubmitEditing={async ({ nativeEvent: { text } }) =>
+            searchForAddress(text)
+          }
+          value={selectedLocation && getFormattedAddress(selectedLocation)}
```

Comment:
this makes the `TextInput` uneditable. You need to handle the `onTextChanged` and share a state between what the user edits and what comes from selecting a spot on the map

Reply:
I managed to do it with useEffect.